core/src/main/scala/mongo4cats/collection/queries/QueryCommand.scala

```diff
@@ -18,209 +18,61 @@ package mongo4cats.collection.queries
 
 import com.mongodb.client.model
 import com.mongodb.client.model.changestream
-import com.mongodb.reactivestreams.client.{AggregatePublisher, ChangeStreamPublisher, DistinctPublisher, FindPublisher}
 import org.bson.{BsonDocument, BsonTimestamp}
 import org.bson.conversions.Bson
-import org.reactivestreams.Publisher
 
-import java.util.concurrent.TimeUnit
 import scala.concurrent.duration.Duration
 
-sealed private[queries] trait QueryCommand[O[_] <: Publisher[_], T] {
-  def run(pub: O[T]): O[T]
+sealed trait DistinctCommand extends Product with Serializable
```

Comment:
Could all of these be `private[queries]`?

Reply:
:+1: